File: data-prep/dataprep-preparators/src/main/java/de/hpi/isg/dataprep/preparators/DeleteProperty.java
package de.hpi.isg.dataprep.preparators;

import de.hpi.isg.dataprep.exceptions.ParameterNotSpecifiedException;
import de.hpi.isg.dataprep.implementation.DeletePropertyImpl;
import de.hpi.isg.dataprep.model.prepmetadata.DeletePropertyMetadata;
import de.hpi.isg.dataprep.model.target.preparator.Preparator;
import de.hpi.isg.dataprep.model.target.preparator.PreparatorImpl;

/**
 * @author Lan Jiang
 * @since 2018/8/20
 */
public class DeleteProperty extends Preparator {

    private String propertyName;

    public DeleteProperty(PreparatorImpl impl) {
        this.impl = impl;
    }

    @Override
    public void buildMetadataSetup() throws ParameterNotSpecifiedException {

    }

    public String getPropertyName() {
        return propertyName;
    }

    public void setPropertyName(String propertyName) {
        this.propertyName = propertyName;
    }
}

File: data-prep/dataprep-preparators/src/main/scala/de/hpi/isg/dataprep/preparators/DeleteProperty.scala
package de.hpi.isg.dataprep.preparators

import de.hpi.isg.dataprep.implementation.DefaultDeletePropertyImpl
import de.hpi.isg.dataprep.model.target.preparator.Preparator

/**
  *
  * @author Lan Jiang
  * @since 2018/8/31
  */
class DeleteProperty(val propertyName : String) extends Preparator {

    this.impl = new DefaultDeletePropertyImpl

    /**
      * This method validates the input parameters of a [[Preparator]]. If succeeds, setup the values of metadata into both
      * prerequisite and toChange set.
      *
      * @throws Exception
      */
    override def buildMetadataSetup(): Unit = {

    }
}
